### DISCUSSION TITLE
How do I get this to appear as an xinput in Ubuntu 20.04?

### DISCUSSION THREAD

**Discussion by baldyman01** [FreeSpacenav/spacenavd#44] [Questions and support]
Hi,

I'm running Ubuntu 20.04 and a Space mouse compact. I'm running the daemon with -d, which from the terminal  output seems fine as far as I can tell -

```
Spacenav daemon 0.6
failed to open config file /etc/spnavrc: No such file or directory. using defaults.
Device detection, parsing /proc/bus/input/devices
found usb device [256f:c635]: "3Dconnexion SpaceMouse Compact" (/dev/input/event7) 
adding device.
device name: 3Dconnexion SpaceMouse Compact
  Number of axes: 6
using device: /dev/input/event7
trying to open X11 display ":1"
   XAUTHORITY=/run/user/1000/gdm/Xauthority

```
From another terminal I'm running 

sudo xhost +
sudo /usr/bin/spnavd_ctl x11 start
signalled spacenavd, it should now start sending X events.

It does not appear in xinput it does not generate events in xev, LED stays lit blue, Blender still works it just fine. Nothing else registers it in any way.

What did I miss?

Cheers

Pete

**Comment by jtsiomb** [comment]
It's not supposed to appear in xinput. Programs such as blender, specifically written for 6dof input, talk to spacenavd to receive events from it. If you want to use the space mouse as an xinput device, you can do that by *not* running spacenavd. But I'm not sure if there are any programs designed to use 6dof input from xinput.

Also running spnavd_ctl is not necessary any more.

**Comment by baldyman01** [comment]
OK, I'm confused. If I don't run your driver the Spacemouse doesn't work at all in anything. No led, no nothing. Is there another driver I'm supposed to use?

**Reply by jtsiomb** [reply]
Maybe it would help if you specified what you are trying to achieve. You did say that blender (and therefore presumably every other program which is written to utilize 6dof input events) works. So what problem are you trying to solve?

The LED is not an indication of functionality. It's just a LED that software can turn on or off. Indeed spacenavd has explicit configuration options for how to handle the LED.

The point of spacenavd is to provide 6dof events to other programs. 3D programs which are written to expect 6dof input, usually are written to expect that input in the form spacenavd provides it, and will not work without it.

Xinput is a generic interface for accessing input devices under X11. When you are *not* running spacenavd, the X server will probably claim any USB 6dof devices it detects, and make them available through the xinput API. Therefore if you have any programs which are written to expect 6dof input from xinput, and not from spacenavd, you should not run spacenavd, and let the X server handle the device. I'm not aware of any such programs however, although it was more common on other UNIX systems (such as IRIX) a long time ago.
